### PR TITLE
Modify the JSON output of kpod history

### PR DESCRIPTION
**Pull request by umohnani8**
The JSON output was being modified to type string, fixed that
to be of the same type as the source
This is better for further processing of the JSON output if needed
Restructured kpod history a bit as well

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@rhatdan @mrunalp @baude @TomSweeneyRedHat PTAL

**Comment by umohnani8** [comment]
During the face to face, I thought we agreed on supporting multiple formats for the `format` flag like JSON, YAML etc. Do we still want to do that?

File: cmd/kpod/history.go [hunk 1]
```diff
@@ -83,226 +94,183 @@ func historyCmd(c *cli.Context) error {
...
 	imgName := args[0]
 
 	opts := historyOptions{
-		image:    imgName,
-		human:    human,
-		truncate: truncate,
-		quiet:    quiet,
-		format:   format,
+		image:   imgName,
+		human:   human,
```
Comment (rhatdan) [inline cmd/kpod/history.go]: If you are cleaning this up, lets get rid of the temporary variables. Any reason to have human?

**Comment by rhatdan** [comment]
@baude was working on this stuff, but this looks like a nice cleanup.
@baude @mheon PTAL

File: cmd/kpod/history.go (hunk 1 above)
Comment (umohnani8) [reply cmd/kpod/history.go]: Yes, human needs to be set to true by default.

File: cmd/kpod/history.go [hunk 2]
```diff
@@ -23,22 +22,34 @@ const (
...
 
-// historyOutputParams stores info about each layer
-type historyOutputParams struct {
-	ID        string `json:"id"`
-	Created   string `json:"created"`
-	CreatedBy string `json:"createdby"`
-	Size      string `json:"size"`
-	Comment   string `json:"comment"`
+// historyTemplateParams stores info about each layer
+type historyTemplateParams struct {
```
Comment (mheon) [inline cmd/kpod/history.go]: I really dislike having two nearly identical structs with the only difference being whether the fields are strings or not, but I can't think of a better way of doing this.

**Comment by mheon** [comment]
Looks like a good change - we should be getting everything moved over to the unified output infrastructure we have written to simplify the code.

File: cmd/kpod/history.go (hunk 2 above)
Comment (umohnani8) [reply cmd/kpod/history.go]: I agree, but with the new formats package I also couldn't see a way around this.

File: cmd/kpod/history.go (hunk 1 above)
Comment (rhatdan) [reply cmd/kpod/history.go]: You can set Human to true by using a BoolTFlag

File: cmd/kpod/history.go [hunk 3]
```diff
@@ -83,226 +94,183 @@ func historyCmd(c *cli.Context) error {
...
-		size        int64
-		img         *storage.Image
-		imageID     string
-		layerAll    []historyOutputParams
-		createdTime string
-		outputSize  string
-	)
-
-	ref, err := is.Transport.ParseStoreReference(store, opts.image)
+// getHistory gets the history of an image and infromation about its layers
```
Comment (TomSweeneyRedHat) [inline cmd/kpod/history.go]: nit infromation to information

File: cmd/kpod/history.go [hunk 4]
```diff
@@ -83,226 +94,183 @@ func historyCmd(c *cli.Context) error {
...
-
-// historyToGeneric makes an empty array of interfaces for output
-func historyToGeneric(params []historyOutputParams) []interface{} {
-	genericParams := make([]interface{}, len(params))
-	for i, v := range params {
-		genericParams[i] = interface{}(v)
+		historyOutput = append(historyOutput, params)
 	}
-	return genericParams
+	return
```
Comment (TomSweeneyRedHat) [inline cmd/kpod/history.go]: I'm beat and am not seeing it, so I'll ask.  When you run 'docker history' you're shown layers that are no longer in play and they're listed as '<missing>'.  How are we handling these layers by ignoring them and not printing them?  Or printing them as though they're not '<missing>'.  I'm not seeing <missing> here, is it just part of the layer or am I missing it?

Comment (umohnani8) [reply cmd/kpod/history.go]: @TomSweeneyRedHat I actually never came across an image with layers that are no longer in play. Could you give me an image that has this and I can test it and see for myself. I just access the `History` struct in the package v1, and the `BlobInfo` for the size of each layer.

File: cmd/kpod/history.go (hunk 1 above)
Comment (umohnani8) [reply cmd/kpod/history.go]: ooh, TIL :)

File: cmd/kpod/history.go (hunk 3 above)
Comment (umohnani8) [reply cmd/kpod/history.go]: fixed.

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@umohnani8 I'll send you a Dockerfile via e-mail that will have "<missing>" layers.

**Comment by umohnani8** [comment]
Can we merge this for now? I will open a new PR to fix the ID issue.

**Comment by rhatdan** [comment]
Sure we just need it to pass tests. :^(

**Comment by rhatdan** [comment]
[retest]

**Comment by umohnani8** [comment]
[retest]

**Comment by umohnani8** [comment]
Rebased

**Comment by umohnani8** [comment]
[retest]

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**